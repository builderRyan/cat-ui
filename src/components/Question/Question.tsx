import * as React from 'react';
import { PureComponent } from 'react';
import Tooltip from '../Tooltip';
import Icon from '../Icon';

export interface QuestionProps {
  /** 类名 */
  className?: string;
  /** 样式 */
  style?: object;
  /** Question hover后显示的内容 */
  tips?: string | React.ReactNode;
}

/**
 * 文字提示
 */
class Question extends PureComponent<QuestionProps, any> {
  render() {
    const { className, style, tips } = this.props;

    return (
      <Tooltip content={tips}>
        <Icon type='question' className={className} style={style} />
      </Tooltip>
    );
  }
}

export default Question;
