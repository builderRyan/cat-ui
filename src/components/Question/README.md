# Question

## Example

```jsx
<Question tips={<div>Question content</div>}  />
```

## API

|   属性    |       类型        | 默认值 |   描述   |
|:---------:|:-----------------:|:------:|:--------:|
| className |     `string`      |   -    |   类名   |
|   style   |     `object`      |   -    | 行内样式 |
|   tips    | `React.ReactNode` |   -    | 提示内容 |
