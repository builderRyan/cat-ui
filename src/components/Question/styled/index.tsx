import styled from 'styled-components'
import { primaryBtnColor } from '../../../styles/theme'

const StyledQuestion = styled.div``;

export { StyledQuestion }
