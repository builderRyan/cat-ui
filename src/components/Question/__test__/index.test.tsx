import * as React from 'react'
import * as Enzyme from 'enzyme'
import { render, mount } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import Question from '../index'

Enzyme.configure({ adapter: new Adapter() })

describe('Question', () => {
  test('renders correctly', () => {
    const wrapper = render(
      <Question>Test</Question>
    )
    expect(wrapper).toMatchSnapshot()
  })

})
