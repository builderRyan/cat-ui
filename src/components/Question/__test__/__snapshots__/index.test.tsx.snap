// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "hmly-popover sc-htpNat dbWHLM",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "hmly-popover sc-bxivhb bUZGRr",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "hmly-tooltip-trigger sc-bdVaJa bDWFJH",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "hmly-icon hmly-icon-question",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
